### DISCUSSION TITLE
Is VSCode able to display inline syntax highlighting different to its surrounding text?

### DISCUSSION THREAD

**Discussion by roke-julian-lockhart** [microsoft/vscode-discussions#937] [Extension Development QnA]
As [`stackoverflow.com/revisions/77519782/3`](https://stackoverflow.com/revisions/77519782/3) states:

<blockQuote>

Say that I'm viewing a Markdown file with the content

~~~.MD
```.IPYNB
{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "Invoke-WebRequest -URI 'https://dub.sh/zza' | ConvertFrom-JSON | Select-Object -ExpandProperty bio"
   ]
  }
 ],
 "metadata": {
  "language_info": {
   "name": "python"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
```
~~~

This contains 3 different types of code:

1.  The
    ~~~.MD
    ```.IPYNB
    ```
    ~~~
    is obviously Markdown;
1.  The proceeding `.JSON` is Jupyter Notebook content (hence the `.IPYNB`); and
1.  The content of the `.IPYNB` file is Markdown containing PowerShell Core (hence the`.PS1`) code.

This is a complex example, but it demonstrates the need for VSCode to be able to intelligently parse these formats to provide syntax highlighting. Is this possible to enable, via an extension or preference?

**Comment by starball5** [comment]
vscode-discussions is only for discussion about extension development. can you clarify how this is about extension development? or did you not realize?

**Reply by roke-julian-lockhart** [reply]
> https://github.com/microsoft/vscode-discussions/discussions/937#discussioncomment-7635848

@starball5, I wasn't aware. What should be used instead?

**Reply by starball5** [reply]
take a look at https://github.com/microsoft/vscode/issues/new/choose and you'll find out. Hint: You've already done it.

**Reply by roke-julian-lockhart** [reply]
> https://github.com/microsoft/vscode-discussions/discussions/937#discussioncomment-7636130

Thanks, @starball5.
